chia/cmds/completion.py

```diff
@@ -0,0 +1,35 @@
+import click
+import os
+import subprocess
+
+SHELLS = ['bash', 'zsh', 'fish']
+
+@click.group()
+def completion() -> None:
+    pass
+
+
+@completion.command()
+@click.option(
+    "-s",
+    "--shell",
+    type=click.Choice(["bash", "zsh", "fish"]),
+    default=os.environ.get('SHELL', '').split('/')[-1],
+    show_default=False,
+    required=True,
+    help="Shell type to generate completion for",
+)
+def generate(shell: str) -> None:
+    """
+    Generate shell completion script
+    """
+    if shell is None:
+        shell = os.environ.get('SHELL', '').split('/')[-1]
+    if shell not in SHELLS:
+        raise click.UsageError(f'Invalid shell type: {shell}')
+    cmd = f'_CHIA_COMPLETE={shell}_source chia'
+    print("Save the following output as '~/.chia-completion."+shell+"' and source it according to your shell:\n")
```

Comment:
We should be able to just get the output for use in automation like `chia completion generate >> place` or to just call on the fly like `eval $(chia completion generate)` or whatever it would be.